elastic/kibana
# [Reporting] Add queue duration metric to event logging

Re-closes https://github.com/elastic/kibana/issues/121468

This PR adds a needed metric to the logged events in reporting: the amount of time that a claimed job waited in `pending` status. Having these metrics will help when using the logs to monitor Reporting, as they will help administrators know if reports are taking more or less time waiting in the queue.

### Testing
1. Enable debug logging for reporting, and use a `json` logging layout. See: https://www.elastic.co/guide/en/kibana/master/logging-configuration.html#json-layout
2. Send 5 requests to generate a Canvas PDF
3. Note the ECS logs generated by reporting, found with the filter: `kibana.reporting.actionType: claim-task`. These logs contain an `event.duration` field that increases in size with each report job:

```
{"event":{"provider":"reporting","duration":2483,"timezone":"America/Phoenix"},"message":"claimed report l12rz8fripwr0763b5df7gd7","kibana":{"reporting":{"actionType":"claim-task","id":"l12rz8fripwr0763b5df7gd7","jobType":"printable_pdf_v2"}},"user":{"name":"elastic"},"ecs":{"version":"8.0.0"},"@timestamp":"2022-03-22T16:35:56.714-07:00","log":{"level":"DEBUG","logger":"plugins.reporting.events"},"process":{"pid":873387},"span":{"id":"7e790bbc264c1dc7"},"trace":{"id":"70a1cee3e7afc4a64cd4f665e58acc25"}}
{"event":{"provider":"reporting","duration":8404,"timezone":"America/Phoenix"},"message":"claimed report l12rzauzipwr0763b5fqne4z","kibana":{"reporting":{"actionType":"claim-task","id":"l12rzauzipwr0763b5fqne4z","jobType":"printable_pdf_v2"}},"user":{"name":"elastic"},"ecs":{"version":"8.0.0"},"@timestamp":"2022-03-22T16:36:05.775-07:00","log":{"level":"DEBUG","logger":"plugins.reporting.events"},"process":{"pid":873387},"span":{"id":"a24fa2ab116fd49c"},"trace":{"id":"1fa21be0036ed2464bd1f0f208ba5a51"}}
{"event":{"provider":"reporting","duration":14715,"timezone":"America/Phoenix"},"message":"claimed report l12rzcw4ipwr0763b59nnxp0","kibana":{"reporting":{"actionType":"claim-task","id":"l12rzcw4ipwr0763b59nnxp0","jobType":"printable_pdf_v2"}},"user":{"name":"elastic"},"ecs":{"version":"8.0.0"},"@timestamp":"2022-03-22T16:36:14.719-07:00","log":{"level":"DEBUG","logger":"plugins.reporting.events"},"process":{"pid":873387},"span":{"id":"ef1405a3810e59d8"},"trace":{"id":"7cf90d35b9fab96cf85bc6af9a9479bb"}}
{"event":{"provider":"reporting","duration":21230,"timezone":"America/Phoenix"},"message":"claimed report l12rzeu4ipwr0763b51pv8y5","kibana":{"reporting":{"actionType":"claim-task","id":"l12rzeu4ipwr0763b51pv8y5","jobType":"printable_pdf_v2"}},"user":{"name":"elastic"},"ecs":{"version":"8.0.0"},"@timestamp":"2022-03-22T16:36:23.754-07:00","log":{"level":"DEBUG","logger":"plugins.reporting.events"},"process":{"pid":873387},"span":{"id":"3e71fe9ffc85da5b"},"trace":{"id":"dde9b58341b5a55fc5d884edab266750"}}
{"event":{"provider":"reporting","duration":27588,"timezone":"America/Phoenix"},"message":"claimed report l12rzgtnipwr0763b5b87xge","kibana":{"reporting":{"actionType":"claim-task","id":"l12rzgtnipwr0763b5b87xge","jobType":"printable_pdf_v2"}},"user":{"name":"elastic"},"ecs":{"version":"8.0.0"},"@timestamp":"2022-03-22T16:36:32.687-07:00","log":{"level":"DEBUG","logger":"plugins.reporting.events"},"process":{"pid":873387},"span":{"id":"2922c9901b398119"},"trace":{"id":"85cf4af50f9e45e0faeb784e8e8b1ec0"}}
```

### Checklist

Delete any items that are not applicable to this PR.
